datadog_checks_base/datadog_checks/base/utils/http.py

```diff
@@ -0,0 +1,248 @@
+# (C) Datadog, Inc. 2019
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import threading
+import warnings
+from collections import OrderedDict
+from contextlib import contextmanager
+
+import requests
+from six import iteritems, string_types
+from six.moves.urllib.parse import urlparse
+from urllib3.exceptions import InsecureRequestWarning
+
+from ..config import is_affirmative
+try:
+    import datadog_agent
+except ImportError:
+    from ..stubs import datadog_agent
+
+STANDARD_FIELDS = {
+    'headers': None,
+    'password': None,
+    'persist_connections': False,
+    'proxy': None,
+    'skip_proxy': False,
+    'ssl_ca_cert': None,
+    'ssl_cert': None,
+    'ssl_ignore_warning': False,
+    'ssl_private_key': None,
+    'ssl_verify': True,
+    'timeout': 10,
+    'username': None,
+}
+DEFAULT_REMAPPED_FIELDS = {
+    # TODO: Remove in 6.13
+    'no_proxy': {'name': 'skip_proxy'},
+}
+PROXY_SETTINGS_DISABLED = {
+    # This will instruct `requests` to ignore the `HTTP_PROXY`/`HTTPS_PROXY`
+    # environment variables. If the proxy options `http`/`https` are missing
+    # or are `None` then `requests` will use the aforementioned environment
+    # variables, hence the need to set each to an empty string.
+    'http': '', 'https': ''
+}
+
+
+class RequestsWrapper(object):
+    __slots__ = ('ignore_ssl_warning', 'persist_connections', 'no_proxy_uris', 'options', '_session')
+
+    # For modifying the warnings filter since the context
+    # manager that is provided changes module constants
+    warning_lock = threading.Lock()
+
+    def __init__(self, instance, init_config, remapper=None):
+        default_fields = dict(STANDARD_FIELDS)
+
+        # Update the default behavior for skipping proxies
+        default_fields['skip_proxy'] = init_config.get('skip_proxy', default_fields['skip_proxy'])
+
+        # Populate with the default values
+        config = {field: instance.get(field, value) for field, value in iteritems(default_fields)}
+
+        # Support non-standard (usually legacy) configurations, for example:
+        # {
+        #     'disable_ssl_validation': {
+        #         'name': 'ssl_verify',
+        #         'default': False,
+        #         'invert': True,
+        #     },
+        #     ...
+        # }
+        if remapper is None:
+            remapper = {}
+
+        remapper.update(DEFAULT_REMAPPED_FIELDS)
+
+        for remapped_field, data in iteritems(remapper):
+            field = data.get('name')
+
+            # Ignore fields we don't recognize
+            if field not in STANDARD_FIELDS:
+                continue
+
+            # Ignore remapped fields if the standard one is already used
+            if field in instance:
+                continue
+
+            # Get value, with a possible default
+            value = instance.get(remapped_field, data.get('default', default_fields[field]))
```

Comment:
What are all the fields the value could have? `name`, `default`, what else? Let's add comments to explain all this.

Reply:
It's in the example above

Reply:
my bad